File: solutions/go/series/1/series.go
package series

// All returns all the series in the given string with the given length.
func All(n int, s string) (res []string) {
	for i := 0; i < len(s)-n+1; i++ {
		res = append(res, s[i:i+n])
	}

	return
}

// UnsafeFirst returns the first substring of the given string with the given length.
func UnsafeFirst(n int, s string) string {
	return s[:n]
}

File: solutions/c/series/1/series.c
#include "series.h"

#include <stdlib.h>
#include <string.h>

slices_t slices(char *input_text, unsigned int substring_length) {
    slices_t res;
    res.substring = calloc(MAX_SERIES_RESULTS, sizeof(char *));
    res.substring_count = 0;

    if (substring_length > MAX_SERIES_LENGTH || substring_length < 1 || !input_text) {
        return res;
    }

    res.substring_count = strlen(input_text) - substring_length + 1;

    for (unsigned int i = 0; i < res.substring_count; i++) {
        res.substring[i] = calloc(MAX_SERIES_LENGTH, sizeof(char));
        strncpy(res.substring[i], input_text + i, substring_length);
    }

    return res;
}